src/libcalamaresui/ViewManager.cpp

```diff
@@ -140,7 +141,7 @@ ViewManager::insertViewStep( int before, ViewStep* step )
 void
 ViewManager::onInstallationFailed( const QString& message, const QString& details )
 {
-    bool shouldOfferWebPaste = false;  // TODO: config var
+    bool shouldOfferWebPaste = Calamares::Branding::instance()->logUploadEnable();
```

Comment:
the simplest thing to do is something like this:

```
QString uploadServerType = Calamares::Branding::instance()->uploadServerType();
bool shouldOfferWebPaste = CalamaresUtils::UploadServers.contains( uploadServerType );
```

where: UploadServers is const QStringList of uploadServerTypes (currently only [ 'fiche' ]); which does not contain the special dummy uploadServerType: 'none'

that eliminates the 'isEnabled' config boolean, the class member, and getter; but still the uploadServerType is unused - FWIW, this is actually suggesting a new CalamaresUtils::UploadServer class

```
CalamaresUtils::UploadServer* uploadServer = CalamaresUtils::UploadServer->instance();
bool shouldOfferLogUpload = uploadServer != nullptr;
```

that is effectively replacing the UploadServers list with a singleton getter, which returns an instance of an UploadServers subclass (eg: FicheUploadServer), instantiated lazily, and selected by the branding config, or nullptr if no UploadServers subclass corresponds to the branding selector

Reply:
I am unable to comprehend what you said after `special dummy uploadServerType: 'none'`
It would be helpful if you can clarify a bit. Sorry for my weak english / tech language.